packages/runtime-core/src/apiDefineComponent.ts

```diff
@@ -290,16 +343,60 @@ export function defineComponent<
   TypeRefs
 >
 
+export function defineComponent<
+  T extends Record<string, any> = {},
+  PropsOptions extends Readonly<ComponentPropsOptions> = {},
+  RawBindings = {},
+  D = {},
+  C extends ComputedOptions = {},
+  M extends MethodOptions = {},
+  Mixin extends ComponentOptionsMixin = ComponentOptionsMixin,
+  Extends extends ComponentOptionsMixin = ComponentOptionsMixin,
+  E extends EmitsOptions = {},
+  EE extends string = string,
+  S extends SlotsType = {},
+>(
+  options: ComponentOptionsBase<
+    PropsOptions,
+    RawBindings,
+    D,
+    C,
+    M,
+    Mixin,
+    Extends,
+    E,
+    EE,
+    ExtractDefaultPropTypes<PropsOptions>,
+    {},
+    string,
+    S
+  > & { __generic?: T },
+): DefineComponentWithGeneric<
+  PropsOptions,
+  RawBindings,
+  D,
+  C,
+  M,
+  Mixin,
+  Extends,
+  E,
+  EE,
+  PublicProps,
+  Readonly<PropsOptions>,
+  ExtractDefaultPropTypes<PropsOptions>,
+  S,
+  T
+>
+
 // implementation, close to no-op
 /*! #__NO_SIDE_EFFECTS__ */
-export function defineComponent(
-  options: unknown,
-  extraOptions?: ComponentOptions,
-) {
-  return isFunction(options)
-    ? // #8326: extend call and options.name access are considered side-effects
-      // by Rollup, so we have to wrap it in a pure-annotated IIFE.
-      /*#__PURE__*/ (() =>
-        extend({ name: options.name }, extraOptions, { setup: options }))()
+export function defineComponent(options: unknown): any {
+  const comp = isFunction(options)
+    ? { setup: options, name: options.name }
```

Comment:
What is the reason for this change? It doesn't seem to be related to types.

Reply:
@edison1105 
The original code had logic to convert function components into objects, but I changed the runtime behavior using Proxy. This was intended to handle props when using Generic Components in TSX.

As you pointed out, this isn't related to types. I realized that by removing extraOptions in this modification, I might have eliminated the ability to extend types.

I'm planning to revert these changes, but is my thought process correct?

Thank you for reviewing multiple times.

Reply:
Thank you for review!
> It looks like the test case is incorrect because it still passes without the changes in this PR.

Does the fact that the test passes without this change mean that the following writing style already exists? 
https://github.com/vuejs/core/blob/e46dfdd771ff0be4ce77a8ef0f03568fe72c564f/packages-private/dts-test/defineComponent.test-d.tsx#L2058-L2061

Or is there a possibility that the original type checking was loose and designed not to throw errors regardless of the writing style?

Reply:
@edison1105 
It has been left here for a long time in a state where I can't proceed with anything. Is there anything I can do?